### DISCUSSION TITLE
Error 401 - Submit signed challenge using KOPS 3.1

### DISCUSSION THREAD

**Discussion by hansjhaase** [Digitaler-Impfnachweis/certification-apis#162] [Q&A]
We stuck in the 4th process: Submitting the signed challenge and public SC-B certificate....
We are using Java11, Libraries like OKHttpd and of course ApiClientCertificate... running our own PVS and KOPS 3.1... 
Below a summary of our process against testing backend.... steps 1-3 are working w/o errors - but step 4 error 401

Any help appreciated! Also examples! Thx

1. Trigger smcb authentication flow
Request{method=GET, url=https://id.certify.demo.ubirch.com/auth/realms/bmg-ti-certify/protocol/openid-connect/auth?client_id=user-access-ti-simulator&redirect_uri=connector%3A%2F%2Fauthenticated&response_type=code&scope=openid, tag=null}
Response{protocol=http/1.1, code=200, message=OK, url=https://id.certify.demo.ubirch.com/auth/realms/bmg-ti-certify/protocol/openid-connect/auth?client_id=user-access-ti-simulator&redirect_uri=connector%3A%2F%2Fauthenticated&response_type=code&scope=openid}

2. Receive Challenge
-> Challenge: f0e46262-2fcb-43e9-aa1f-3098a1e155cc
-> Location: https://id.certify.demo.ubirch.com/auth/realms/bmg-ti-certify/login-actions/authenticate?session_code=Jp1N0iOf_-6W_IDQ-qy6qFa0nDfSOnp0XK96qXE2dwQ&execution=a4825c72-879a-4f96-8ed5-d1c1b0e6bf70&client_id=user-access-ti-simulator&tab_id=9sS-SiOtZG4&auth_session_id=89293991-8ab7-4d1a-813a-33fff47aa13b

3. Perform Signature operation with SMCB
Call ExternalAuthenticate
Return ExternalAuthenticate
Call ReadCardCertificate
Return ReadCardCertificate
Certificate (Base64): MIIEpzCCA4+gAwIBAgIEXEou7TANBgkqhkiG9w0BAQsFADCBhzEjMCEGA1UEAwwaQy5FSEVYLkhVTUFOLUNBMSBURVNULU9OTFkxNTAzBgNVBAsMLGVIZWFsdGhFeHBlcnRzLUNBIGRlciBUZWxlbWF0aWtpbmZyYXN0cnVrdHVyMRwwGgYDVQQKDBNlSGVhbHRoRXhwZXJ0cyBHbWJIMQswCQYDVQQGEwJERTAeFw0yMTA3MTMxNTIzMDhaFw0yNjEyMzAyMzAwMDBaMGUxGjAYBgNVBAMMEVRlc3QgUHJheGlzIFZhbGlkMRwwGgYDVQQFExMxLXNtY2ItZG9jdG9yLXZhbGlkMRwwGgYDVQQKDBNlSGVhbHRoRXhwZXJ0cyBHbWJIMQswCQYDVQQGEwJERTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAKkKY4Kq2ZWRCZny+8lr67/+tnruAJpxjDPDzPINl7YC6XtLr4X/nPVhAm/7A4OhIw3ZD80un2xG1zT0sk/SloYNy7E5CHrwbXyTQFsgs/KY6Yv6qtGHhzJgxZgCxh51dEqNABNXjeufOPybifhpfTuOLBPWOMZsm7+Z0YvHaqcWSMIe+Z+B6tahnU3pmIBQtVm2PCRQ4y7J2IaGJJWDmSOu374gZZ3PD+dWDhB6nhI49v5tkEIaLH22yaGw1qWgnOtvCBSckkcKq8acT/0/p/CluRZ6f/KT9bt2FHtOdvq3pAKpsKqiQvtKXFR2Xsblq+Qy5dUvFxAl9fdNO7uBvwsCAwEAAaOCATowggE2MB8GA1UdIwQYMBaAFO6yj0G41h0TlmFKVvRS2nPV9dVjMB0GA1UdDgQWBBQNSxtEyraUYexl3/apjyRGvNxfcjAMBgNVHRMBAf8EAjAAMA4GA1UdDwEB/wQEAwIFoDATBgNVHSUEDDAKBggrBgEFBQcDAjAgBgNVHSAEGTAXMAoGCCqCFABMBIEjMAkGByqCFABMBE0wUgYIKwYBBQUHAQEERjBEMEIGCCsGAQUFBzABhjZodHRwOi8vb2NzcC5laGV4LmRlOjI1NjAvZWpiY2EvcHJpdmF0ZS93ZWIvc3RhdHVzL29jc3AwSwYFKyQIAwMEQjBAMD4wPDA6MDgwFgwUQmV0cmllYnNzdMOkdHRlIEFyenQwCQYHKoIUAEwEMhMTMS1zbWNiLWRvY3Rvci12YWxpZDANBgkqhkiG9w0BAQsFAAOCAQEAdM2jLVswdvF3he0lp5fdwLKKLmdom3dnepLRqJbh8NBWeLGlYAZMfMbvGzkK4zjhYNdcm+paQZgIjChvo1NA1uuvSWcighlFbFaX+xk8aYyZvPYigCYa6W2AIWqiVO8PFhjQoh7tlr/NzBVHMpYo7a3Y6hihHlNZajTCXzPi2wEPz46HbagHmHboYMVTpfk58MRYhyc+qB0GfD/JDnWbIC/dOhKodt0SUT82OVp0KQxetcTvSJik5fNm06BEGjJ+UlMa05leqZ7Ohr3rocwqcZImIuLzaRm7i7PrZyrGpifuXPqcVD9fWa/z2Na1a4/XVeUJ01TsVUKcFQLNC9x18g==
Signed Challenge (Base64): kZDCbaPmX38vxNDkXqvlizwvHet9678TozANBw/W9Y+mWaOClR8tf35lkDqWMTpYVHtH2FdOr5hd0EpvX/uhvgcdtoA342RmwCmcps9OfzLmFSMgUkMOgGO3dGLA1Pm+zSe8cbFeCmQITSFlhgIK444pXb3aJL1zEk3/A5ax3pWgSM+vRTPa579cZ0bPK3N1z3yJHH1mBA1YC+/kvf4UEBe6O91NukhTEoroQ0feMHJApyYu4GEEQytChJVD42rHEhX07Yk8nCHjyIU/y32miorCtJmRBtmjF48MFAThlP61vhFUpf/jj9EZ+S/c+XFW9kLLLWJsbTG+Y30aE6mVAA==

4. Submit Signet challenge and SMCB public certificate
Request{method=GET, url=https://id.certify.demo.ubirch.com/auth/realms/bmg-ti-certify/login-actions/authenticate?session_code=Jp1N0iOf_-6W_IDQ-qy6qFa0nDfSOnp0XK96qXE2dwQ&execution=a4825c72-879a-4f96-8ed5-d1c1b0e6bf70&client_id=user-access-ti-simulator&tab_id=9sS-SiOtZG4&auth_session_id=89293991-8ab7-4d1a-813a-33fff47aa13b, tag=null}
Response{protocol=**http/1.1, code=401**, message=Unauthorized, url=https://id.certify.demo.ubirch.com/auth/realms/bmg-ti-certify/login-actions/authenticate?session_code=Jp1N0iOf_-6W_IDQ-qy6qFa0nDfSOnp0XK96qXE2dwQ&execution=a4825c72-879a-4f96-8ed5-d1c1b0e6bf70&client_id=user-access-ti-simulator&tab_id=9sS-SiOtZG4&auth_session_id=89293991-8ab7-4d1a-813a-33fff47aa13b}

**Comment by richSpiegel** [comment]
Hi there,
we are facing the same problem! Have you fixed that issue yourself?
However we use this endpoint: https://id.ru.impfnachweis.info/auth/realms/bmg-ti-certify/protocol/openid-connect/auth

Regards - Richard
